components/script/dom/xmlhttprequest.rs

```diff
@@ -786,12 +786,12 @@ impl XMLHttpRequest {
     fn change_ready_state(&self, rs: XMLHttpRequestState) {
         assert!(self.ready_state.get() != rs);
         self.ready_state.set(rs);
+
         let global = self.global.root();
-        let event = Event::new(global.r(),
-                               atom!("readystatechange"),
-                               EventBubbles::DoesNotBubble,
-                               EventCancelable::Cancelable);
-        event.fire(self.upcast());
+        self.upcast::<EventTarget>().fire_simple_event_params("readystatechange",
```

Comment:
We don't want to fire a simple event here, this calls https://dom.spec.whatwg.org/#concept-event-fire .


...which seems to do the same thing, not sure why there are two. But leave this unchanged for now